runtime/lua/tohtml.lua

```diff
@@ -1291,7 +1291,29 @@ local function opt_to_global_state(opt, title)
   if opt.font then
     fonts = type(opt.font) == 'string' and { opt.font } or opt.font --[[@as (string[])]]
   elseif vim.o.guifont:match('^[^:]+') then
-    table.insert(fonts, vim.o.guifont:match('^[^:]+'))
+    -- Example:
+    -- Input: "Font,Escape\,comma, Ignore space after comma"
+    -- Output: { "Font","Escape,comma","Ignore space after comma" }
+    local str, char_pre = '', ''
+    for _, char in ipairs(vim.split(vim.o.guifont:match('^[^:]+'), '')) do
```

Comment:
IIUC this is splitting on empty string as a way to inspect each character? And most of the logic is trying to deal with escaped commas (`\,`) ? Why not use `vim.gsplit` instead of `ipairs(vim.split)` ?

This seems to work:

```lua
local fonts = {}
local prev = ''
for name in vim.gsplit(vim.o.guifont:match('^[^:]+'), ',', {trimempty=true}) do
  if vim.endswith(name, '\\') then
    prev = prev .. vim.trim(name:sub(1, -2) .. ',')
  elseif vim.trim(name) ~= '' then
    table.insert(fonts, ('"%s%s"'):format(prev or '', vim.trim(name)))
    prev = ''
  end
end
```

Reply:
Yes, that’s right. That implementation didn’t occur to me...
As for the quotation marks, since opt.font also needs to be enclosed in them, I think it would be better to do that below rather than here.

Reply:
> since opt.font also needs to be enclosed in them

can we just change line 1292 to:

    fonts = { ('"%s"'):format(type(opt.font) == 'string' and opt.font or opt.font[1]) }

Reply:
With this code, if multiple fonts are specified in opt.font, only the first item is applied. Is this the expected behavior?

Reply:
see different suggestion: https://github.com/neovim/neovim/pull/30105/files#r1742248186

Reply:
yes. the point is that logic is only needed for that if-branch, there is no reason to repeat it for both branches.